LibreNMS/OS/Beagleboard.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Beagleboard.php
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2017 Tony Murray
+ * @author     Tony Murray <murraytony@gmail.com>
+ */
+
+namespace LibreNMS\OS;
+
+use LibreNMS\Interfaces\Discovery\OSDiscovery;
+use LibreNMS\OS;
+
+class Beagleboard extends OS implements
+    OSDiscovery
+{
+    /**
+     * Retrieve basic information about the OS / device
+     */
+    public function discoverOS(): void
+    {
+        $device = $this->getDeviceModel();
+        $device->version = implode(' ', array_slice(explode(' ', trim(snmp_get($this->getDevice(), 'NET-SNMP-EXTEND-MIB::nsExtendOutputFull."distro"', '-Osqnv'), '"')), 1));
+        $device->hardware = trim(snmp_get($this->getDevice(), 'nsExtendOutputFull."hardware"', '-Osqnv'), '"');
```

Comment:
Can't you combine these snmp calls with snmp_get_multi()?

Reply:
I could (I think), but then still have to process them separately after. Was thinking this was clearer ... thoughts?

Reply:
It's more efficient with a multi query, see for example https://github.com/librenms/librenms/blob/master/LibreNMS/OS/Qnap.php

Reply:
I can see that, from an snmp call perspective. But as I'm not using MIBs and names, the return value seems messed up with snmp_getnext_multi => the results become the array keys, not values? Data below, for example,
```
Array
(
    [BeagleBone AI] =>
    [18] =>
)
```

OK to leave it as is?

Thanks!

Reply:
It should work, see like https://github.com/librenms/librenms/blob/master/LibreNMS/OS/AviatWtm.php#L50

Reply:
Yep, this makes sense - but like you can see above, I'm getting "odd" results when not using named OID's (i.e. from an MIB). Rather, I'm using extend, so items like nsExtendOutputFull."hardware". It's not seeming to like that 😞.

I have tried the approach in both of the above - no joy.

Reply:
Fixed! Had to use a slightly different function, and change the snmp options. But working now, Thanks!